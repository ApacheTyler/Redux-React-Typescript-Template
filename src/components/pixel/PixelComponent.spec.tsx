import * as React from 'react'
import { shallow, mount, render } from 'enzyme'

import { createStore } from 'redux'

import Pixel from './PixelComponent'


describe('<Pixel />', () => {

  it('renders', () => {
    expect(shallow(
      <Pixel placeholderText='Welcome To Redux!' message='Hello World!' buttonText='Set Text' buttonClicked={() => {}}/>
    )).toMatchSnapshot()
  });

})
