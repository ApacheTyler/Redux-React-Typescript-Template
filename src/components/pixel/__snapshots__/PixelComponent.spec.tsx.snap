// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pixel /> renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h1>
        Hello World!
    </h1>
    <input
        name="messageText"
        placeholder="Welcome To Redux!"
        type="text"
    />
    <button
        onClick={[Function]}
    >
        Set Text
    </button>
</div>,
  "nodes": Array [
    <div>
      <h1>
            Hello World!
      </h1>
      <input
            name="messageText"
            placeholder="Welcome To Redux!"
            type="text"
      />
      <button
            onClick={[Function]}
      >
            Set Text
      </button>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Pixel
        buttonClicked={[Function]}
        buttonText="Set Text"
        message="Hello World!"
        placeholderText="Welcome To Redux!"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Pixel {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "buttonClicked": [Function],
          "buttonText": "Set Text",
          "message": "Hello World!",
          "placeholderText": "Welcome To Redux!",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h1>
                    Hello World!
          </h1>
          <input
                    name="messageText"
                    placeholder="Welcome To Redux!"
                    type="text"
          />
          <button
                    onClick={[Function]}
          >
                    Set Text
          </button>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h1>
                    Hello World!
          </h1>
          <input
                    name="messageText"
                    placeholder="Welcome To Redux!"
                    type="text"
          />
          <button
                    onClick={[Function]}
          >
                    Set Text
          </button>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Pixel
    buttonClicked={[Function]}
    buttonText="Set Text"
    message="Hello World!"
    placeholderText="Welcome To Redux!"
/>,
}
`;
